dspace-spring-rest/src/test/java/org/dspace/app/rest/EntityTypeRestRepositoryIT.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.sql.SQLException;
+
+import org.dspace.app.rest.matcher.EntityTypeMatcher;
+import org.dspace.app.rest.test.AbstractEntityIntegrationTest;
+import org.dspace.content.EntityType;
+import org.dspace.content.service.EntityTypeService;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class EntityTypeRestRepositoryIT extends AbstractEntityIntegrationTest {
+
+
+    @Autowired
+    private EntityTypeService entityTypeService;
+
+    @Test
+    public void findAllEntityTypesSizeTest() throws SQLException {
+        assertEquals(7, entityTypeService.findAll(context).size());
+    }
+
+    @Test
+    public void findPublicationEntityTypeTest() throws SQLException {
+        String type = "Publication";
```

Comment:
@abollini : is this comment on the correct line?  It doesn't make sense to me.  Maybe I'm misunderstanding, but it looks like this code is already just checking one value?  If you are talking about all the methods following this, then I'd disagree slightly.  These are all more of "Unit Tests" (so I agree we should move them), but checking each EntityType is a unit test that verifies each type was loaded from configuration properly

Reply:
@tdonohue I think I disagree:  the unit under test doesn't seem to be a configuration loader, so checking for proper loading here would not be a unit test, but a kind of integration test.  In a unit test, we should already know that a valid configuration useful for testing is loaded, because the test code itself provides it.

There is probably no point in integration-testing the loading of configuration.  There should be a thorough unit test of the loader, depending only on code external to the product.  If tests pass, then we can rely on the loaded configuration to be what we give externally.  The loader's unit-testing configuration should test the loader only, and probably will be useless for other testing.

The configuration data themselves, if complex enough, should have a validator that operates at run time, not a test.  The *validator* should have tests, and if it passes, then we can rely on the configuration to be valid.

Reply:
Good point.  In any case, it looks like this has been cleaned up in the outstanding PR #2418